File: frontend/src/dtos/broker/brokerDto.ts
import type { BrokerAccessDto } from './brokerAccessDto'
import type { PostgresConnectionDto } from './postgresConnectionDto'
import type { RabbitMQConnectionDto } from './rabbitMQConnectionDto'

export interface BrokerDto {
  id: string
  userId: string
  createdByUserId: string
  accessList: BrokerAccessDto[]
  system: string
  name: string
  rabbitMQConnection: RabbitMQConnectionDto
  postgresConnection: PostgresConnectionDto
  createdAt: string
  updatedAt: string
  syncedAt: string
  deletedAt: string
}

File: backend/ResQueue/ResQueue/Dtos/Broker/BrokerDto.cs
using ResQueue.Dtos.Broker;

namespace ResQueue.Dtos;

public record BrokerDto(
    string Id,
    string CreatedByUserId,
    List<BrokerAccessDto> AccessList,
    string System,
    string Name,
    PostgresConnectionDto? PostgresConnection,
    DateTime CreatedAt,
    DateTime UpdatedAt,
    DateTime? SyncedAt,
    DateTime? DeletedAt
);